test/unit/matrix/test_panel.cpp

```diff
@@ -378,17 +379,31 @@ void testShrink(const config_t& cfg, const comm::CommunicatorGrid& comm_grid) {
       const LocalTileIndex idx(coord, k);
       dlaf::internal::transformLiftDetach(dlaf::internal::Policy<Backend::MC>(), setTile,
                                           panel.readwrite_sender(idx), counter++);
-      const auto& tile = panel.read(idx).get();
+
+      auto panel_tile_f = panel.read(idx);
+      const auto& panel_tile = panel_tile_f.get();
+      auto matrix_tile_f = matrix.read(idx);
+      const auto& matrix_tile = matrix_tile_f.get();
+
+      // TODO: Comment about why the gets are up there above the SCOPED_TRACE?
+      // This was again a case of gtest expecting SCOPED_TRACE to be destroyed
+      // on the same thread it was constructed.
```

Comment:
Yes, a comment is welcomed.

Reply:
Please have a look, I've added a comment both about the dangling reference and `SCOPED_TRACE`. I didn't reproduce the comment in the for loop just below which does the same but can add a note there as well if you'd like.

Reply:
Good! Isn't it worth to mention gtest in the second part?
I think is better to add at least a small note in the for loop as well.

Reply:
I think both the dangling reference and the gtest comment apply in the second part as well. What about I add just a short comment to see the comment above since it applies there as well?
```
// See comment in for loop above. The same problems about dangling references and SCOPED_TRACE apply here as well.
```
?

Reply:
Sorry for the delay, updated! I think this can be merged for testing on #554 now.